01_fib/main_test.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMainOutput(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+	// When
+	main()
+
+	// Then
+	expected := "0112358\n"
```

Comment:
Also test `fib` (unit tests) and corner cases (limit: -1, 0, 1, very large number).